docs/config.md

```diff
@@ -200,6 +200,27 @@ can override the default style with the following keys:
 ui.diff.format = "git"
 ```
 
+#### Color-words diff options
+
+In color-words diffs, changed words are displayed inline by default. Because
+it's difficult to read a diff line with many removed/added words, there's a
+threshold to switch to traditional separate-line format.
+
+* `max-inline-alternation`: Maximum number of removed/added word alternation to
+  inline. For example, `<added> <added>` sequence has 1 alternation, so the line
+  will be inline if `max-inline-alternation >= 1`. `<added> <removed> <added>`
+  sequence has 3 alternation.
+
+  * `0`: disable inlining
+  * `1`: inline removes-only or adds-only lines
+  * `2`, `3`, ..: inline up to `2`, `3`, .. alternation
+  * `-1`: inline all lines (default)
+
+```toml
+[diff.color-words]
+max-inline-alternation = 3
```

Comment:
I'm wondering if this config could be something simpler.

Perhaps

```toml
inline = true
inline-max-complexity = 3
```

For the docs, we could first explain the intuition (your 0,1,... examples), and then have a section with the details.

Another thought:

Reply:
I assume we'll settle on some better default, and users rarely customize this parameter. It's hard to explain what the "complexity" (or "alternation") means.

Reply:
Marked this parameter as experimental. If we find a better default and formalization, we can deprecate the current one.